lib/block-supports/duotone.php

```diff
@@ -550,6 +541,81 @@ static function () use ( $filter_svg, $selector ) {
 	);
 }
 
+
+
+/**
+ * Outputs all necessary SVG for duotone filters and CSS for classic themes.
+ */
+add_action(
+	'wp_footer',
+	static function () {
+
+		foreach ( WP_Duotone_Gutenberg::$output as $filter_data ) {
+
+			// SVG will be output on the page later.
+			$filter_svg = gutenberg_get_duotone_filter_svg( $filter_data );
+
+			echo $filter_svg;
+
+			// This is for classic themes - in block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
+
+			if ( ! wp_is_block_theme() ) {
+				wp_add_inline_style( 'core-block-supports', 'body{' . WP_Duotone_Gutenberg::get_css_custom_property_declaration( $filter_data ) . '}' );
+			}
+
+			global $is_safari;
+			if ( $is_safari ) {
+				duotone_safari_rerender_hack( $filter_data['selector'] );
+			}
+		}
+	}
+);
+
+/**
+ * Appends the used global style duotone filter CSS Vars to the inline global styles CSS
+ */
+add_action(
+	'wp_enqueue_scripts',
+	static function() {
+
+		if ( empty( WP_Duotone_Gutenberg::$output ) ) {
+			return;
+		}
+
+		$duotone_css_vars = '';
+
+		foreach ( WP_Duotone_Gutenberg::$output as $filter_data ) {
+			if ( ! array_key_exists( $filter_data['slug'], WP_Duotone_Gutenberg::$global_styles_presets ) ) {
+				continue;
+			}
+
+			$duotone_css_vars .= WP_Duotone_Gutenberg::get_css_custom_property_declaration( $filter_data );
+		}
+
+		if ( ! empty( $duotone_css_vars ) ) {
+			wp_add_inline_style( 'global-styles', 'body{' . $duotone_css_vars . '}' );
+		}
+	},
```

Comment:
Maybe the functions for this and the `wp_footer` above could be moved into `WP_Duotone_Gutenberg` too.

If we do, we may be able to make `WP_Duotone_Gutenberg::$output` private to the class as well to minimize the public API.

Reply:
If we change this, we'll also have to move `gutenberg_render_duotone_support` into the class as well.

Reply:
Yeah, it's completely reliant on WP_Duotone_Gutenberg::$output, so let's let the class handle it.

Reply:
> If we change this, we'll also have to move gutenberg_render_duotone_support into the class as well.

I moved the `wp_footer` and `wp_enqueue_scripts` actions into the WP_Duotone_Gutenberg class and it appears to be working fine. Why do we need to move `gutenberg_render_duotone_support` as well?

Reply:
Moved it anyways, btw.

Reply:
> Why do we need to move `gutenberg_render_duotone_support` as well? Is it to be able to make it private?

It was using `WP_Duotone_Gutenberg::$output`, so yes, it would have to be part of the class if `$output` is private.